### PR TITLE
feat: Create a Testcontainer runtime for Process Testing

### LINKED ISSUE #18994: [CPT] Create a Testcontainer runtime for Process Testing
## Description

In Camunda Process Test, we want to run the process tests on Camunda based on [Testcontainers](https://java.testcontainers.org/). The first task is to create the Testcontainer runtime and provide all required containers for Camunda. Take inspiration from the community project: [zeebe-test-container](https://github.com/camunda-community-hub/zeebe-test-container).

Why Testcontainer? 
- We need a Testcontainer to run Elasticsearch
- We have the following benefits of running Camunda inside a Testcontainer:
    - Avoid dependency version conflicts from the Camunda libraries and the process application
    - Independent from the Java version 
    - Independent from the Spring framework

## Tasks
- Create a new module `camunda-process-test`
- Create Testcontainers for Zeebe, Operate, Tasklist, and Elasticsearch
- Allow the configuration of:
    - Environment Variables
    - Logging
    - Exposed ports
    - Docker image 
    - Camunda version

### PR DESCRIPTION
**Pull request by saig0**
## Description

- Create a new Maven module for the new Camunda Process Test library for Java: `camunda-process-test-java`
- Create a runtime for process tests based on Docker Testcontainers with containers for:
    - Elasticsearch - used from Testcontainers
    - Zeebe¹
    - Operate¹ 
    - Tasklist¹
- Expose certain configuration options
- Align the Docker image version with the testing library
    - By default, use the same Camunda image version as the testing library 
    - By default, use the same Elasticsearch image version as the Elasticsearch client in Camunda
    - Use a properties file with Maven placeholders to resolve the versions 
- The implementation is inspired by the community project: [zeebe-test-container](https://github.com/camunda-community-hub/zeebe-test-container). :coin: 

¹ These containers will be removed later by a single Camunda image with all components.

## Related issues

closes #18994

### CODE REVIEW

**Review by nicpuppa — COMMENTED** [review]
🔧 Should we create a separate CI workflow for this new project ?

File: testing/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Testing
+
+Camunda's testing libraries for processes and process applications.
+
+- [Camunda Process Test for Java](camunda-process-test-java)
```
[the remark this answers is not in the document]
Comment (saig0) [reply testing/README.md:5]: Not for other programming languages but for the Spring integration.

File: testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/containers/OperateContainer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    applyDefaultConfiguration();
+  }
+
+  private void applyDefaultConfiguration() {
+    withNetwork(Network.SHARED)
+        .waitingFor(newDefaultWaitStrategy())
+        .addExposedPorts(ContainerRuntimePorts.OPERATE_REST_API);
+  }
+
+  public OperateContainer withZeebeGrpcApi(final String zeebeGrpcApi) {
```
[the remark this answers is not in the document]
Comment (saig0) [reply testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/containers/OperateContainer.java:45]: Not for now. I will expose the endpoint if needed.

File: testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/containers/OperateContainer.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy.Mode;
+import org.testcontainers.utility.DockerImageName;
+
+public class OperateContainer extends GenericContainer<OperateContainer> {
+
+  private static final Duration DEFAULT_STARTUP_TIMEOUT = Duration.ofMinutes(1);
+  private static final String OPERATE_READY_ENDPOINT = "/actuator/health/readiness";
```
[the remark this answers is not in the document]
Comment (saig0) [reply testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/containers/OperateContainer.java:32]: We could but I don't see a reason to expose these constants outside the container. `ContainerRuntimeDefaults` contains relevant defaults for the runtime configuration.

File: testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/runtime/ContainerRuntimeVersionUtil.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  public static final Logger LOGGER = LoggerFactory.getLogger(ContainerRuntimeVersionUtil.class);
+
+  public static final String VERSION_PROPERTIES_FILE = "/camunda-container-runtime.properties";
+
+  public static final String PROPERTY_NAME_CAMUNDA_VERSION = "camunda.version";
+  public static final String PROPERTY_NAME_ELASTICSEARCH_VERSION = "elasticsearch.version";
+
+  public static final String CAMUNDA_VERSION_SNAPSHOT = "SNAPSHOT";
+  public static final String ELASTICSEARCH_VERSION_DEFAULT = "8.13.0";
```
[the remark this answers is not in the document]
Comment (saig0) [reply testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/runtime/ContainerRuntimeVersionUtil.java:37]: Yes, this is done in the `camunda-container-runtime.properties` file. It contains the placeholder `${version.elasticsearch}`. During the Maven build, the placeholder is replaced with the concrete version by Maven.

[the remark this answers is not in the document]
Comment (nicpuppa) [reply testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/runtime/ContainerRuntimeVersionUtil.java:37]: I missed that 🙇

**Review by nicpuppa — APPROVED** [review]
Adding the CI should be done in a separate PR (I think). LGTM 🚀

**Pull request merged**